vcd/resource_vcd_vapp.go

```diff
@@ -286,9 +288,18 @@ func resourceVcdVAppUpdate(d *schema.ResourceData, meta interface{}) error {
 
 	if d.HasChange("metadata") {
 		oraw, nraw := d.GetChange("metadata")
-		metadata := oraw.(map[string]interface{})
-		for k := range metadata {
-			task, err := vapp.DeleteMetadata(k)
+		ometadata := oraw.(map[string]interface{})
```

Comment:
Issue with variable naming
```suggestion
		oldMetaData := oldRaw.(map[string]interface{})
```

Reply:
Fixed.